src/main.rs

```diff
@@ -39,46 +52,20 @@ fn main() {
     let mut state = entity::worldstate::WorldState::new_with_map(all_entities);
 
     let mut i = 1;
-    while i <= MAX_STEPS {
-        println!("-----------------STEP {:?}-----------------", i);
-        state.step(DT);
-        i += 1;
-    }
+    let mut is_open = true;
 
     thread::spawn(|| {
         play_sick_beats();
     });
-    setup_window();
-}
-
-fn setup_window() {
-    let mut window = three::Window::new("Gentle AP Physics RS");
-    window.scene.background = three::Background::Color(0xC6F0FF);
-
-    // TODO: replace this sample code with our own `entities`
-    // BEGIN SAMPLE CODE
-    let vertices = vec![
-        [-0.5, -0.5, -0.5].into(),
-        [0.5, -0.5, -0.5].into(),
-        [0.0, 0.5, -0.5].into(),
-    ];
-    let geometry = three::Geometry::with_vertices(vertices);
-    let material = three::material::Basic {
-        color: 0xFFFF00,
-        map: None,
-    };
-    let mesh = window.factory.mesh(geometry, material);
-    window.scene.add(&mesh);
-
-    let center = [0.0, 0.0];
-    let yextent = 1.0;
-    let zrange = -1.0 .. 1.0;
-    let camera = window.factory.orthographic_camera(center, yextent, zrange);
 
-    while window.update() {
-        window.render(&camera);
+    while i <= MAX_STEPS && is_open {
+        println!("-----------------STEP {:?}-----------------", i);
+        state.step(DT);
+        is_open = render(&mut window, &camera, &state.all_entities());
+        i += 1;
+        thread::sleep(time::Duration::from_millis(500));
```

Comment:
500 ms is a little high, do we want to lower this?

Reply:
changed to `50`